Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2TransportNIOPosix.swift

```diff
@@ -21,13 +21,35 @@ import NIOExtras
 import NIOPosix
 
 extension HTTP2ServerTransport {
+  /// A NIOPosix-backed implementation of a server transport.
   @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
   public struct Posix: ServerTransport {
     private let address: GRPCHTTP2Core.SocketAddress
     private let config: Config
     private let eventLoopGroup: MultiThreadedEventLoopGroup
     private let serverQuiescingHelper: ServerQuiescingHelper
 
+    private let listeningAddressPromise:
+      _LockedValueBox<EventLoopPromise<GRPCHTTP2Core.SocketAddress>>
```

Comment:
This is slightly annoying, but I backed this by a box containing the promise because I didn't want to make `listen` a mutating func, and I need to mutate the promise when we exit `listen`'s scope.